webui/react/src/components/LogViewerTimestamp.tsx

```diff
@@ -409,63 +382,63 @@ const LogViewerTimestamp: React.FC<Props> = ({
         ),
       });
     }
-
-    // update log timestamp range
-    let max: string|null = null;
-    let min: string|null = null;
-    newLogs.forEach(log => {
-      if (!log.time) return;
-      if (!max || log.time > max) max = log.time;
-      if (!min || log.time < min) min = log.time;
-    });
-    setLogTimestampRange(prev => ({
-      max: (prev.max && max && dayjs(prev.max).isAfter(dayjs(max))) ? prev.max : max,
-      min: (prev.min && min && dayjs(prev.min).isBefore(dayjs(min))) ? prev.min : min,
-    }));
   }, [ container ]);
 
   const clearLogs = useCallback(() => {
+    setIsScrollOn({ bottom: true, top: true });
     setIsLastReached(false);
     setLogs([]);
-    setLogTimestampRange({
-      max: null,
-      min: null,
-    });
   }, []);
 
+  const handleFilterChange = useCallback((newFilters: TrialLogFiltersInterface) => {
+    clearLogs();
+    setFilter(newFilters);
+  }, [ clearLogs ]);
+
+  const handleScrollToTop = useCallback(() => {
+    clearLogs();
+    setDirection(DIRECTIONS.OLDEST);
+  }, [ clearLogs ]);
+
+  const handleEnableTailing = useCallback(() => {
+    clearLogs();
+    setDirection(DIRECTIONS.TAILING);
+  }, [ clearLogs ]);
+
+  const handleDownload = useCallback(() => {
+    if (onDownloadClick) onDownloadClick();
+  }, [ onDownloadClick ]);
+
   /*
    * Fetch filters data.
    */
   useEffect(() => {
+    const canceler = new AbortController();
+
     consumeStream(
-      onFetchLogFilter(historyCanceler),
+      onFetchLogFilter(canceler),
       event => setFilterOptions(event as LogViewerTimestampFilter),
     );
 
-    return () => historyCanceler.abort();
-  }, [ historyCanceler, onFetchLogFilter ]);
+    return () => canceler.abort();
+  }, [ onFetchLogFilter ]);
 
   /*
-   * Load first batch of logs (tail or oldest).
+   * Watch Log tail (api follow).
    */
   useEffect(() => {
+    if (direction !== DIRECTIONS.TAILING) return;
+
     const canceler = new AbortController();
-    clearLogs();
 
     let buffer: TrialLog[] = [];
     const throttleFunc = throttle(THROTTLE_TIME, () => {
       addLogs(buffer);
       buffer = [];
     });
 
-    const fetchArgs = (
-      direction === DIRECTIONS.TAILING
-        ? onFetchLogTail(filter, canceler)
-        : onFetchLogAfter({ ...filter, timestampAfter: dayjs(0) }, canceler)
-    );
-
     consumeStream(
-      fetchArgs,
+      onFetchLogTail(filter, canceler),
```

Comment:
**question:** should we have this `useEffect` return? I remember you mentioning this can cause some problems but without it, it means the tailing stream will continue to exist and stream. So if you switch between oldest mod and tailing mode back and forth a lot, it will accumulate multiple streaming calls without aborting any of them.
```
return () => canceler.abort();
```

Reply:
I think that's effect is returning, probably you checked the diff and + / - lines added some confusion?
<img width="562" alt="Schermata 2021-01-25 alle 11 20 55" src="https://user-images.githubusercontent.com/5413702/105692974-6b9c2a00-5eff-11eb-8bdf-c0e12e88a076.png">